LibreNMS/Device/Sensor.php

```diff
@@ -358,9 +358,12 @@ private static function fetchSnmpData($device, $sensors)
             $snmp_data = array_merge($snmp_data, $multi_data);
         }
 
-        // deal with string values that may be surrounded by quotes
+        // deal with string values that may be surrounded by quotes, scientific number format and remove non numerical characters
         array_walk($snmp_data, function (&$oid) {
-            $oid = trim($oid, '"') + 0;
+            $oid = preg_replace('/[^-0-9.e]/', '', $oid);
```

Comment:
Hello Murrant. If this is the wrong place than please help me find the right place. I think this might be a great place to generically solve the issues as described below. Furthermore the two times preg_replace are really needed:
The first one removes all characters that should not be in a number or scientific number. So it filters thinks like additional things like V for voltage, A for amperage, C for celcius, and also things like "TX=7123.123Mhz' or 'dB; are cleaned. But the 'e' stays as it might be a scientific number. Then after trying to convert the scientific number to a numeric value it might still have been no scientific number so then we need to clean the value in the second preg_replace.

Reply:
Hmm, this parsed `2.1E-5` incorrectly for example.